src/test/java/com/pusher/rest/CryptoTest.java

```diff
@@ -0,0 +1,88 @@
+package com.pusher.rest;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.pusher.rest.util.Crypto;
+import com.pusher.rest.util.EncryptedPayload;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Base64;
+
+import static org.junit.Assume.assumeTrue;
+
+public class CryptoTest {
+    private static final Gson BODY_SERIALISER = new GsonBuilder()
+            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+            .create();
+
+    private Crypto pc = null;
+    @Before
+    public void initialize() {
+        if(Crypto.cryptoAvailable()) {
+             pc = new Crypto("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", BODY_SERIALISER);
+        }
```

Comment:
So if it's not available, what happens to these tests? They all fail?

Reply:
Crypto cannot be constructed without this method being true, hence this check.

This instance is shared among the tests hence its position in this @before annotated method, but this does have the slight downside that it isn't obvious that the tests will be skipped if crypto is not available (see `assumeTrue(Crypto.cryptoAvailable());`). Crazily, assumeTrue will just pass the test if crypto is not available, even though Junit has the concept of a skipped test. If crypto is available, the test executes in full.

Reply:
Crypto cannot be constructed without this method being true, hence this check.

This instance is shared among the tests hence its position in this @before annotated method, but this does have the slight downside that it isn't obvious that the tests will be skipped if crypto is not available (see `assumeTrue(Crypto.cryptoAvailable());`). Crazily, assumeTrue will just pass the test if crypto is not available, even though Junit has the concept of a skipped test. If crypto is available, the test executes in full.

Reply:
Ah right. Yeah that makes sense. Is it not possible to put the `assumeTrue`s in the `@Before`?

Reply:
No, I tried it that way but it seems to need to be in the definition of the test.

Reply:
Fair enough.